components/date-time/time.js

```diff
@@ -122,43 +182,103 @@ class TimePicker extends Component {
-					<Button
-						isDefault
-						className="components-time-picker__am-button"
-						isToggled={ am === 'AM' }
-						onClick={ this.updateAmPm( 'AM' ) }
-					>
-						{ __( 'AM' ) }
-					</Button>
-					<Button
-						isDefault
-						className="components-time-picker__pm-button"
-						isToggled={ am === 'PM' }
-						onClick={ this.updateAmPm( 'PM' ) }
-					>
-						{ __( 'PM' ) }
-					</Button>
-				</div> }
+				<fieldset key="date-time-form" className="date-time-form">
+					<div key="month-field" className="input-container month">
+						<select
+							id="date-time-form--month-field"
+							aria-labelledby={ __( 'Month' ) }
+							name="month"
+							value={ month }
+							onChange={ this.onChangeMonth }
+							onBlur={ this.updateMonth }
+						>
+							<option value="01">{ __( 'January' ) }</option>
+							<option value="02">{ __( 'February' ) }</option>
+							<option value="03">{ __( 'March' ) }</option>
+							<option value="04">{ __( 'April' ) }</option>
+							<option value="05">{ __( 'May' ) }</option>
+							<option value="06">{ __( 'June' ) }</option>
+							<option value="07">{ __( 'July' ) }</option>
+							<option value="08">{ __( 'August' ) }</option>
+							<option value="09">{ __( 'September' ) }</option>
+							<option value="10">{ __( 'October' ) }</option>
+							<option value="11">{ __( 'Nobember' ) }</option>
+							<option value="12">{ __( 'December' ) }</option>
+						</select>
+					</div>
+					<div key="day-field" className="input-container day">
+						<input
+							id="date-time-form--day-field"
+							className="components-time-picker__input day"
+							aria-labelledby={ __( 'Day' ) }
+							type="text"
+							name="day"
+							value={ day }
+							onChange={ this.onChangeDay }
+							onBlur={ this.updateDay }
+						/>
+					</div>
+					<div key="year-field" className="input-container year">
+						<input
+							id="date-time-form--year-field"
+							className="components-time-picker__input"
+							aria-labelledby={ __( 'Year' ) }
+							type="text"
+							name="year"
+							value={ year }
+							onChange={ this.onChangeYear }
+							onBlur={ this.updateYear }
+						/>
+					</div>
+				</fieldset>
+				<fieldset className="date-time-form">
+					<input
+						id="date-time-form--hour-field"
+						className="components-time-picker__input"
+						aria-labelledby={ __( 'Hours' ) }
+						type="text"
+						name="hours"
+						value={ hours }
+						onChange={ this.onChangeHours }
+						onBlur={ this.updateHours }
+					/>
+					<span
+						className="components-time-picker__separator"
+						aria-hidden="true">:</span>
+					<input
+						id="date-time-form--minute-field"
+						className="components-time-picker__input"
+						aria-labelledby={ __( 'Minutes' ) }
+						type="text"
+						value={ minutes }
+						onChange={ this.onChangeMinutes }
+						onBlur={ this.updateMinutes }
+					/>
+					{ is12Hour && <fieldset aria-labelledby={ __( 'AM or PM' ) }>
```

Comment:
fieldset elements shouldn't use `aria-labelledby`; instead they should use a `<legend>` element to give the group a name. In this case though I'm not sure I'd use a fieldset in the first place. I'd say the two buttons "AM" and "PM" are clear enough.

Reply:
I aggree!